[2022-12-12 12:07:08,683] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: daily_Paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [queued]>
[2022-12-12 12:07:08,694] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: daily_Paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [queued]>
[2022-12-12 12:07:08,694] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2022-12-12 12:07:08,695] {taskinstance.py:1357} INFO - Starting attempt 1 of 2
[2022-12-12 12:07:08,695] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2022-12-12 12:07:08,706] {taskinstance.py:1377} INFO - Executing <Task(PythonOperator): get_joke_task> on 2022-12-11 00:00:00+00:00
[2022-12-12 12:07:08,711] {standard_task_runner.py:52} INFO - Started process 199 to run task
[2022-12-12 12:07:08,723] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'daily_Paper', 'get_joke_task', 'scheduled__2022-12-11T00:00:00+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/morning-paper.py', '--cfg-path', '/tmp/tmppn5n10a1', '--error-file', '/tmp/tmp_60i5x4q']
[2022-12-12 12:07:08,726] {standard_task_runner.py:80} INFO - Job 3: Subtask get_joke_task
[2022-12-12 12:07:08,805] {task_command.py:369} INFO - Running <TaskInstance: daily_Paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [running]> on host cbcd6542cb8e
[2022-12-12 12:07:08,890] {taskinstance.py:1571} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=akash.khamkar40@gmail.com
AIRFLOW_CTX_DAG_OWNER=akashk
AIRFLOW_CTX_DAG_ID=daily_Paper
AIRFLOW_CTX_TASK_ID=get_joke_task
AIRFLOW_CTX_EXECUTION_DATE=2022-12-11T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-12-11T00:00:00+00:00
[2022-12-12 12:07:09,896] {taskinstance.py:1889} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 171, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 189, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/morning-paper.py", line 32, in get_joke
    if res.status_code != 200 or joke_dict['error'] == True:
UnboundLocalError: local variable 'joke_dict' referenced before assignment
[2022-12-12 12:07:09,907] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=daily_Paper, task_id=get_joke_task, execution_date=20221211T000000, start_date=20221212T120708, end_date=20221212T120709
[2022-12-12 12:07:09,915] {standard_task_runner.py:97} ERROR - Failed to execute job 3 for task get_joke_task (local variable 'joke_dict' referenced before assignment; 199)
[2022-12-12 12:07:09,931] {local_task_job.py:156} INFO - Task exited with return code 1
[2022-12-12 12:07:09,957] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2022-12-12 14:19:33,761] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: daily_paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [queued]>
[2022-12-12 14:19:33,778] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: daily_paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [queued]>
[2022-12-12 14:19:33,779] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2022-12-12 14:19:33,779] {taskinstance.py:1357} INFO - Starting attempt 1 of 2
[2022-12-12 14:19:33,780] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2022-12-12 14:19:33,797] {taskinstance.py:1377} INFO - Executing <Task(PythonOperator): get_joke_task> on 2022-12-11 00:00:00+00:00
[2022-12-12 14:19:33,802] {standard_task_runner.py:52} INFO - Started process 1712 to run task
[2022-12-12 14:19:33,807] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'daily_paper', 'get_joke_task', 'scheduled__2022-12-11T00:00:00+00:00', '--job-id', '103', '--raw', '--subdir', 'DAGS_FOLDER/morning-paper.py', '--cfg-path', '/tmp/tmprxj_b8cm', '--error-file', '/tmp/tmpn5ei01a9']
[2022-12-12 14:19:33,809] {standard_task_runner.py:80} INFO - Job 103: Subtask get_joke_task
[2022-12-12 14:19:33,929] {task_command.py:369} INFO - Running <TaskInstance: daily_paper.get_joke_task scheduled__2022-12-11T00:00:00+00:00 [running]> on host cbcd6542cb8e
[2022-12-12 14:19:34,085] {taskinstance.py:1571} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=akash.khamkar40@gmail.com
AIRFLOW_CTX_DAG_OWNER=akashk
AIRFLOW_CTX_DAG_ID=daily_paper
AIRFLOW_CTX_TASK_ID=get_joke_task
AIRFLOW_CTX_EXECUTION_DATE=2022-12-11T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-12-11T00:00:00+00:00
[2022-12-12 14:19:35,111] {logging_mixin.py:115} INFO - {'error': False, 'category': 'Misc', 'type': 'single', 'joke': "Two reasons I don't give money to homeless people.\n1) They are going to spend it all on drugs and alcohol\n2) I am going to spend it all on drugs and alcohol.", 'flags': {'nsfw': False, 'religious': False, 'political': False, 'racist': False, 'sexist': False, 'explicit': False}, 'id': 72, 'safe': False, 'lang': 'en'}
[2022-12-12 14:19:35,115] {python.py:173} INFO - Done. Returned value was: Two reasons I don't give money to homeless people.
1) They are going to spend it all on drugs and alcohol
2) I am going to spend it all on drugs and alcohol.
[2022-12-12 14:19:35,178] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=daily_paper, task_id=get_joke_task, execution_date=20221211T000000, start_date=20221212T141933, end_date=20221212T141935
[2022-12-12 14:19:35,207] {local_task_job.py:156} INFO - Task exited with return code 0
[2022-12-12 14:19:35,235] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
